dotnet/roslyn
# Add remote for Core to package

How-to test change in this PR:

1. Go here https://devdiv.visualstudio.com/DevDiv/_packaging?_a=feed&feed=VSIDE-RealSigned-Release%40Local, and search for “microsoft.servicehub.host.clr”. Download the host nuget package. The newest version of the package should be fine as long as you’re using a recent-ish build of VS. Unzip the package and under the tools directory, copy the coreClr directory to your VS install under Common7/ServiceHub/Hosts.

2. Open Common7\ServiceHub\Hosts\coreClr\ServiceHub.Host.CLR.runtimeconfig.json, and change it to targeting 3.1
{
  "runtimeOptions": {
    "tfm": "netcoreapp3.1",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "3.1.0"
    }
  }
}

3. Build this branch and deploy the vsix

4. Set these environmental variables to control the behavior of Roslyn remote service:

RoslynServiceHubCore : set to no empty to use .net core servicehub
RoslynServiceHubCoreDebug : set to no empty to launch debugger and break before we start initializing remote service. This would help you debug
Launch VS, open a C# project and wait a bit for servicehub to spin-up.

Address #45535
